packages/app/features/dashboard/beat/CancelOfferModal.tsx

```diff
@@ -0,0 +1,156 @@
+import { Entry, useUpdatePricingMutation } from "app/api/graphql";
+import { Button, Modal, Pressable, Text } from "app/design-system";
+import { tw } from "app/design-system/tailwind";
+import { useCallback, useState } from "react";
+import X from "app/ui/icons/x";
+import { useToast } from "react-native-toast-notifications";
+import { useErrorReport } from "app/hooks/useErrorReport";
+import { useWalletConnectClient } from "app/provider/WalletConnect";
+import { WalletConnectModal } from "app/ui/modal/WalletConnectModal";
+import { sellOffersUrl } from "app/hooks/useUserOffers";
+import { useSWRConfig } from "swr";
+import { useRecoilValue } from "recoil";
+import { userAtom } from "app/state/user";
+import { getEntryOfferUrl } from "app/hooks/useEntryOffer";
+
+type Props = {
+  visible: boolean;
+  entry: Entry;
+  offerId: string;
+  hideModal: () => void;
+};
+
+export const CancelConfirmationModal = ({
+  visible,
+  hideModal,
+  entry,
+  offerId,
+}: Props) => {
+  const [loading, setLoading] = useState<boolean>(false);
+  const [message, setMessage] = useState<string | undefined>();
+  const [updatePricing] = useUpdatePricingMutation();
+  const toast = useToast();
+  const reportError = useErrorReport();
+  const { signAndSubmitXdr, session, connect } = useWalletConnectClient();
+  const [uri, setUri] = useState<string>("");
+  const [walletConnectModalVisible, setWalletConnectModalVisible] =
+    useState<boolean>(false);
+
+  const user = useRecoilValue(userAtom);
+  const { mutate } = useSWRConfig();
+
+  const revalidateOffers = () => {
+    mutate(getEntryOfferUrl(entry.code, entry.issuer));
+    mutate(sellOffersUrl(user?.publicKey, entry.issuer, entry.code));
+  };
+
+  const handelSubmit = useCallback(async () => {
+    setLoading(true);
+    const { data } = await updatePricing({
+      variables: {
+        id: entry.id!,
+        equityForSale: 0,
+        price: 0,
+        forSale: false,
+        offerID: offerId,
+      },
+    });
+
+    if (data?.updatePricing?.success) {
+      if (data.updatePricing.submitted) {
+        setLoading(false);
+        hideModal();
+        toast.show("You have successfully cancelled an offer", {
+          type: "success",
+        });
+        revalidateOffers();
+      } else if (data.updatePricing.xdr) {
+        setMessage("Sign and submit transaction in your wallet");
+        const xdr = data.updatePricing.xdr;
+
+        try {
```

Comment:
you can have try-catch block for the entire function in case updatePricing mutation fails